### PR TITLE
update roadmap section

### PR DESCRIPTION
**Pull request by alixander**
rather than use this as a duplicate of TODOs, just link to the github issues and use the space for longform writing on priorities

### CODE REVIEW

File: docs/tour/future.md [hunk 1]
```diff
@@ -1,21 +1,189 @@
...
+
+### How success is measured
+
+- If you're making a diagram for a large audience (e.g. a blog, company all-hands), would
+  you use D2 to make it?
+
+One deciding factor is the time it takes to get to a presentable diagram. Presentable
+doesn't mean full of vibrant colors and shadows and design fluff. It does mean consistent
+colors that increase clarity, on-brand palette of company colors, distinct shapes and
+arrow -- properties that make a professional diagram you'd be proud of presenting.
```
Comment (nhooyr) [inline docs/tour/future.md:93]:
```suggestion
arrows -- properties that make a professional diagram you'd be proud of presenting.
```

File: docs/tour/future.md [hunk 2]
```diff
@@ -1,21 +1,189 @@
...
+customization takes. I wish the bar here could be that customizing via text can get better
+than the GUI, but I don't believe this to be a possible outcome. If you want a slightly
+different shade of green, are you going to edit the hex code by typing it? Or would you
+rather point and click from a gradient. If you want to change the style of 3 objects
+you're looking at, you can text-search for the names of each of those 3 objects and change
+them individually. But most of us would rather use the superior input device for precision
+targeting -- a mouse.
+
+While getting the customization experience to be GUI-level is merely aspirational, D2 can
+aim to be at least as good as an existing, battle-tested text-based styling language: CSS.
```
Comment (nhooyr) [inline docs/tour/future.md:105]:
```suggestion
aim to be at least as good as the existing, battle-tested text-based styling language: CSS.
```

File: docs/tour/future.md [hunk 3]
```diff
@@ -1,21 +1,189 @@
...
+
+To further support composability, D2 has a built-in API for manipulating the AST at a
+high-level (more intelligent than just CRUD on AST nodes). This lets anyone build up and
+edit a diagram programmatically. The goal isn't to enable a specific use case, but rather
+unforeseen ones for the infinite workflows out there.
+
+D2's plugin system further makes the language extensible. These allow you to add "hooks"
+to stages of D2 compilation. For example, while not core to D2, a hypothetical set of
+plugins can add a styled border, add your signature/credits, make everything look
+hand-drawn, then increase contrast for accessibility.
```
Comment (nhooyr) [inline docs/tour/future.md:134]: Link to sketch PR.

File: docs/tour/future.md [hunk 4]
```diff
@@ -1,21 +1,189 @@
...
+
+### Plan
+
+1. D2's plugin system is still in its early stages, and should allow deeper hooks into
+parts of compilation.
+1. Complete editor integrations. Currently, syntax highlighting is supported. But a
+feature-complete integration for VSCode for example would allow rendering to its built-in
+browser, autoformat, call out to LSP functions to refactor, and more.
+1. Build out imports/exports. Currently, D2 can take in D2 files and export to SVGs. It
+should have a transpiler to import other popular text-to-diagram languages, and output to
```
Comment (nhooyr) [inline docs/tour/future.md:158]: Add leading spaces to align these so they're easier to read.

File: docs/tour/future.md [hunk 5]
```diff
@@ -1,21 +1,189 @@
...
+
+For each of these, there are survey questions that are used to measure progress.
+
+## Layouts
+
+In some instances, you need a diagram to match the exact image in your head. These
+workflows will always require a design tool, with a drag-and-drop GUI. In other cases, you
+don't care about exact match, or you can't even picture the end result. The requirements
+for these types of diagrams are more relaxed --  it only needs to satisfy a set of
+  constraints. These are the types of diagrams that an algorithm can get right --
```
Comment (nhooyr) [inline docs/tour/future.md:26]: Random leading spaces

File: docs/tour/future.md [hunk 6]
```diff
@@ -1,21 +1,189 @@
...
+
+[3] For now, this is closed-source. It's free to download and evaluate. To learn more,
+visit [https://terrastruct.com/tala](https://terrastruct.com/tala).
+
+[4] If you still miss the hand-drawn aesthetic, D2 has just the thing:
+https://github.com/terrastruct/d2/pull/91
+
+[5] Actually, gopls single-handedly brings my machine back to 2005 speeds with how much
+CPU it consumes. Not the fault of vim-go though!
+
```
Comment (nhooyr) [inline docs/tour/future.md:188]: Random trailing newlines.

File: docs/tour/future.md [hunk 7]
```diff
@@ -1,21 +1,189 @@
...
+[2] MermaidJS seems to have implemented support for containers as well, but it's not
+widely released and their live editor still won't allow it.
+
+[3] For now, this is closed-source. It's free to download and evaluate. To learn more,
+visit [https://terrastruct.com/tala](https://terrastruct.com/tala).
+
+[4] If you still miss the hand-drawn aesthetic, D2 has just the thing:
+https://github.com/terrastruct/d2/pull/91
+
+[5] Actually, gopls single-handedly brings my machine back to 2005 speeds with how much
```
Comment (nhooyr) [inline docs/tour/future.md:186]: God bless you. I absolutely despise gopls...

**Review by nhooyr — CHANGES_REQUESTED** [review]
well written. i like justification for GUI and text cross compat. sometimes mouse + GUI is best, despite what we programmers sometimes like to believe in regards to our keyboards.

File: docs/tour/future.md (hunk 2 above)
Comment (alixander) [reply docs/tour/future.md:105]: sounds weird, i think "an" is the right word here

File: docs/tour/future.md (hunk 5 above)
Comment (alixander) [reply docs/tour/future.md:26]: yeah idk why `gq` adds these

**Review by nhooyr — APPROVED** [review]

**Pull request merged**